crates/ra_proc_macro/src/process.rs

```diff
@@ -46,22 +47,25 @@ impl Drop for Process {
 
 impl Process {
     fn run(
-        process_path: PathBuf,
+        path: PathBuf,
         args: impl IntoIterator<Item = impl AsRef<OsStr>>,
     ) -> io::Result<Process> {
-        let child = Command::new(&process_path)
-            .args(args)
+        let args = args.into_iter().map(|s| s.as_ref().into()).collect();
+
+        let child = Command::new(&path)
+            .args(&args)
             .stdin(Stdio::piped())
             .stdout(Stdio::piped())
             .stderr(Stdio::null())
             .spawn()?;
```

Comment:
What about extracting child creation in a separate function or event making restart virtually this?
```rust
fn restart(self) -> io::Result<Process> {
    Process:run(self.path, self.args)
}
```

Reply:
By the way, if restart fails, you end up with a killed child, but I am not sure if this can break smth

Reply:
The problem here is Process implement `Drop`, we have to clone `path` and `args` here. I wonder it is worth or not.

Reply:
Correct, it will be dropped and it will kill the child in its drop impl, however any of the proposed variants are equal to me, at least making a helper function would be clearer.